docs/maui/layouts/DockLayout.md

```diff
@@ -0,0 +1,121 @@
+
+### C#
+
+To set the docking position from C#, use `DockLayout.SetDockPosition(IView, DockPosition)` to apply the attached `DockPosition` property.
+
+```csharp
+var button = new Button { Text = "Top", HeightRequest = 50 };
+DockLayout.SetDockPosition(button, DockPosition.Top);
+```
+
+Alternatively, `DockLayout` contains a fluent-style `Add(IView, DockPosition)` method that adds a child view to the layout container and sets it's docking position.
+
+```csharp
+var dockLayout = new DockLayout();
+var button = dockLayout.Add(new Button { Text = "Top", HeightRequest = 50 }, DockPosition.Top);
+```
+
+Using the overload, a `DockLayout` can be constructed conveniently in C# as shown here:
+
+```csharp
+using CommunityToolkit.Maui.Layouts;
+
+var page = new ContentPage
+{
+	Content = new DockLayout
+	{
+		{ new Button { Text = "Top", HeightRequest = 50 }, DockPosition.Top },
+		{ new Button { Text = "Bottom", HeightRequest = 70 }, DockPosition.Bottom },
+		{ new Button { Text = "Left", WidthRequest = 80 }, DockPosition.Left },
+		{ new Button { Text = "Right", WidthRequest = 90 }, DockPosition.Right },
+        { new Button { Text = "Center" } },
+	}
+};
+```
+
+Note: `DockPosition.None` is the default and can be omitted.
+
+## Customizing a DockLayout
+
+A `DockLayout` container supports arbitrary `Padding` as well as several `DockLayout`-specific properties for customization. An example in XAML with all available options is given here:
+
+### XAML
+
+```xml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:mct="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+             x:Class="MyProject.MyContentPage">
+
+    <mct:DockLayout HeightRequest="400"
+                    WidthRequest="600"
+                    Padding="10,20,30,40"
+                    VerticalSpacing="10"
+                    HorizontalSpacing="15"
+                    ShouldExpandLastChild="False">
+        ...
+    </mct:DockLayout>
+    
+</ContentPage>
+```
+
+## Properties
+
+|Property  |Type  |Description  |
+|---------|---------|---------|
+| `Padding` | `Thickness` | Gets or sets the padding around the layout container (inherited from `Layout`). |
+| `HorizontalSpacing` | `double` | Gets or sets the **horizontal** spacing between docked views. |
+| `VerticalSpacing` | `double` | Gets or sets the **vertical** spacing between docked views. |
+| <td colspan=2>`HorizontalSpacing` and `VerticalSpacing` is applied between neighboring views in the `DockLayout`. For example, `HorizontalSpacing` is added between *Left*, *None*, and *Right* views, but also between neighboring views in the same `DockPosition` such as multiple views docked to the *Left*. `VerticalSpacing` is rendered between vertically stacked views in *Top*, *None*, and *Bottom* positions.</td> |
```

Comment:
I'll have to defer to @jfversluis here and let us know how this renders in the preview pages

Reply:
Table looks good, the image in the top seems to be broken though: 
![image](https://user-images.githubusercontent.com/939291/200332531-40d53c5b-319c-46a2-8b55-398948ccf1f8.png)

Reply:
Sorry can you check the preview again, I believe @profix898 has tried using a relative path to the image now